apps-rendering/src/server/insertNewsletter.ts

```diff
@@ -0,0 +1,226 @@
+				targetPlace - paragraphsFromPreviousElements,
+			text: current.text,
+		};
+	}
+
+	return null;
+};
+
+/**
+ * Splits a Text BodyElement into to parts after a give number of
+ * nodes in the html and puts another BodyElement between them.
+ *
+ * @param textElement a Text BodyElement from the original Item Body
+ * @param elementToInsert A Body element to place inside the text
+ * @param splitPoint the index of the node in the textElement to split before
+ * @returns a Tupple of three BodyElements
+ */
+const splitTextElementAndInsertBetweenParts = (
+	textElement: Text,
+	elementToInsert: BodyElement,
+	splitPoint = 1,
+): [Text, BodyElement, Text] => {
+	// uses a clone of the original doc to avoid mutations
+	const childNodeArray = Array.from(
+		textElement.doc.cloneNode(true).childNodes,
+	);
+
+	const newDocs: [DocumentFragment, DocumentFragment] = [
+		JSDOM.fragment(''),
+		JSDOM.fragment(''),
+	];
+
+	childNodeArray.slice(0, Math.floor(splitPoint)).forEach((element) => {
+		newDocs[0].appendChild(element);
+	});
+
+	childNodeArray.slice(Math.floor(splitPoint)).forEach((element) => {
+		newDocs[1].appendChild(element);
+	});
+
+	return [
+		{
+			kind: ElementKind.Text,
+			doc: newDocs[0],
+		},
+		elementToInsert,
+		{
+			kind: ElementKind.Text,
+			doc: newDocs[1],
+		},
+	];
+};
+
+/**
+ * Tries to insert a NewsletterSignUp BodyElement into the middle of an article.
+ *
+ * @param body the array of BodyElement Results for the item
+ * @param newsletter The newsletter to create a NewsletterSignUp BodyElement for
+ * @param internalShortId The Optional id for the Item
+ * @returns a new array of BodyElement Results with the NewsletterSignUp BodyElement
+ * inserted in the middle of the article, or the unmodifed body if the placement logic
+ * does not identify a suitable place
+ * @sideEffect logs a warning if the placement logic does not identify a suitable place
+ */
+const tryToinsertNewsletterIntoStandard = (
+	body: Array<Result<string, BodyElement>>,
+	newsletter: Newsletter,
+	internalShortId: Option<string>,
+): Array<Result<string, BodyElement>> => {
+	const insertPlace = findWhereToInsert(body);
+
+	if (!insertPlace) {
+		logger.warn(
+			`Unable to find suitable place for NewsletterSignupBlockElement`,
+			internalShortId.kind === OptionKind.Some
+				? internalShortId.value
+				: '[undefined internalShortId]',
+		);
+		return body;
+	}
+
+	const { bodyResultIndex, bodyResultParagraphIndex, text } = insertPlace;
+	const textWithNewsletter = splitTextElementAndInsertBetweenParts(
+		text,
+		{
+			kind: ElementKind.NewsletterSignUp,
+			...newsletter,
+		},
+		bodyResultParagraphIndex || 1, // don't place at zero (before first paragraph)
+	);
+
+	return [
+		...body.slice(0, bodyResultIndex),
+		...textWithNewsletter.map(wrapInResult),
+		...body.slice(bodyResultIndex + 1), // +1 so the original Text that was split is left out
+	];
+};
+
+export const insertNewsletterIntoItem = (
+	item: Item,
```

Comment:
I'll expand on this in the review comment, but I think we could simplify this module by acceping an array of body elements here, like:

```suggestion
	item: BodyElement[],
	format: ArticleFormat
```

and using format to determine whether to insert the newsletter